### PR TITLE
[deckhouse-controller] DeckhouseRelease without annotations is not reconciled

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Description
Fixed a bug related to the fact that the state of the release object was not updated

## Why do we need it, and what problem does it solve?
Close https://github.com/deckhouse/deckhouse/issues/9823

## Why do we need it in the patch release (if we do)?
The issue is relevant for the current production

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: fix
summary: Fixed a bug related to the fact that the state of the release object was not updated
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by libmonsoon-dev** [comment]
/e2e/abort static;Static;containerd;1.30 11034556546 11034556546-1-con-1-30 /sys/deckhouse-oss/install:pr9838 astra@192.168.199.122

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've recognized your '/e2e/abort' command and started the workflow...

File: deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1/deckhouse_release.go [hunk 1]
```diff
@@ -25,15 +25,15 @@ import (
...
 	DeckhouseReleaseGVK = schema.GroupVersionKind{
 		Group:   SchemeGroupVersion.Group,
 		Version: SchemeGroupVersion.Version,
-		Kind:    "DeckhouseRelease",
+		Kind:    DeckhouseReleaseKind,
+	}
+	DeckhouseReleaseGVR = schema.GroupVersionResource{
+		Group:    SchemeGroupVersion.Group,
+		Version:  SchemeGroupVersion.Version,
+		Resource: DeckhouseReleasesKind,
```
Comment (yalosev) [inline deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1/deckhouse_release.go:36]: It's not a kind, it's a resource. It must be lowercase and plural form
```suggestion
		Resource: "deckhousereleases",
```

Comment (libmonsoon-dev) [reply deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1/deckhouse_release.go]: Fixed. Actually this structure is not used

**Review by miklezzzz — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9896](https://github.com/deckhouse/deckhouse/pull/9896) to the branch [release-1.64](https://github.com/deckhouse/deckhouse/tree/release-1.64) successful!

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit f482a320027e19cc154a53c20cf3c8bb524169ee to the branch [release-1.63](https://github.com/deckhouse/deckhouse/tree/release-1.63) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/11597179827) for details.

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit f482a320027e19cc154a53c20cf3c8bb524169ee to the branch [release-1.63](https://github.com/deckhouse/deckhouse/tree/release-1.63) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/11597179827) for details.

**Pull request merged**